src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -3694,13 +3700,28 @@ export class ResultsViewPageStore {
     });
 
     readonly treatmentsInStudies = remoteData<Treatment[]>({
-        await: () => [this.studyIds],
+        await: () => [this.molecularProfilesInStudies],
         invoke: async () => {
-            return internalClient.fetchTreatmentsUsingPOST({
-                treatmentFilter: {
-                    studyIds: this.studyIds.result!,
-                } as TreatmentFilter,
-            });
+            const treatmentResponseProfiles = this.molecularProfilesInStudies.result.filter((profile) => {
+                return profile.genericAssayType === GenericAssayTypeConstants.TREATMENT_RESPONSE;
+            })
+            const treatmentResponseProfileIds = treatmentResponseProfiles.map(profile => profile.molecularProfileId);
+            const treatmentsMeta: GenericAssayMeta[] = await client.fetchGenericAssayMetaDataUsingPOST(
```

Comment:
ts should be infer type from client method return type.  if it's not, something is wrong.